src/test/java/com/kodilla/ecommercee/repository/ProductTestSuite.java

```diff
@@ -0,0 +1,139 @@
+package com.kodilla.ecommercee.repository;
+
+import com.kodilla.ecommercee.domain.*;
+import com.kodilla.ecommercee.repository.*;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@Transactional
+
+public class ProductTestSuite {
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Test
+    public void productAddingTest() {
+
+        //Given
+        Product product1 = new Product("Shoes", "Super comfortable running shoes", new BigDecimal(199.99));
+        Product product2 = new Product("Suit", "100% wool business suit", new BigDecimal(1199.00));
+        Product product3 = new Product("Dress", "Posh evening dress", new BigDecimal(799.00));
+
+
+        //When
+        productRepository.save(product1);
+        productRepository.save(product2);
+        productRepository.save(product3);
+
+        List<Product> productList = (List<Product>) productRepository.findAll();
+
+        long productId = product1.getProductId();
+        String product1Name = product1.getProductName();
+
+        product3.setProductName("Evening Dress");
+        productRepository.save(product3);
+        String newProduct3Name = product3.getProductName();
+
+        //Then
+        assertEquals(3, productList.size());
+        assertEquals("Suit", productList.get(1).getProductName());
+        assertEquals("Shoes", product1Name);
+        assertEquals("Evening Dress", newProduct3Name);
+        assertNotEquals(0, productId);
+
+        //Clean-up
+        productRepository.delete(product1);
+        productRepository.delete(product2);
+        productRepository.delete(product3);
+    }
+
+    @Test
+    public void productDeletingTest() {
+
+        //Given
+        Product product1 = new Product("Shoes", "Super comfortable running shoes", new BigDecimal(199.99));
+        Product product2 = new Product("Suit", "100% wool business suit", new BigDecimal(1199.00));
+        Product product3 = new Product("Dress", "Posh evening dress", new BigDecimal(799.00));
+
+
+        //When
+        productRepository.save(product1);
+        productRepository.save(product2);
+        productRepository.save(product3);
+
+        List<Product> productListBeforeDeletion = (List<Product>) productRepository.findAll();
```

Comment:
Po co to rzutowanie?

Reply:
Wcześniej w `ProductRepository` nie było nadpisanej metody findAll() i chyba dlatego nie wychodziło - teraz ją dodałem i testy wychodzą bez rzutowania.